password-cli/app/Main.hs

```diff
@@ -0,0 +1,209 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Main (main) where
+
+import Control.Exception (bracket_)
+import Control.Monad (unless, void, when)
+import Data.Version (showVersion)
+import qualified Data.Password.Argon2 as Argon2
+import Data.Password.Bcrypt (PasswordCheck (..))
+import qualified Data.Password.Bcrypt as Bcrypt
+import qualified Data.Password.PBKDF2 as PBKDF2
+import qualified Data.Password.Scrypt as Scrypt
+import Data.Password.Types
+import Data.Text (Text)
+import qualified Data.Text.IO as T
+import Options.Applicative
+import Paths_password_cli (version)
+import System.Environment (getArgs)
+import System.Exit (exitFailure)
+import System.IO (IOMode(ReadMode), hSetEcho, stdin, stderr, withFile)
+
+main :: IO ()
+main =
+  execParserPure defaultPrefs cliOpts . defaultHelp <$> getArgs
```

Comment:
Oh... just saw this while going through `optparse-applicative`

[`showHelpOnEmpty`](https://hackage.haskell.org/package/optparse-applicative-0.18.1.0/docs/Options-Applicative.html#v:showHelpOnEmpty)

:sweat_smile: No need to do this manually, it seems.

```haskell
customExecParser (prefs showHelpOnEmpty) cliOpts
```